graphql-server/src/schemas/neo4j.ts

```diff
@@ -336,109 +342,30 @@ function buildResolvers(
-              collectionYear
-              genePanel
-              igoComplete
-              igoRequestId
-              importDate
-              investigatorSampleId
-              libraries
-              oncotreeCode
-              preservation
-              primaryId
-              qcReports
-              sampleClass
-              sampleName
-              sampleOrigin
-              sampleType
-              sex
-              species
-              tissueLocation
-              tubeId
-              tumorOrNormal
-              hasStatusStatuses {
-                validationReport
-                validationStatus
-              }
-            }
-            requestsHasSample {
-              igoRequestId
-              igoProjectId
-              genePanel
-              dataAnalystName
-              dataAnalystEmail
-              dataAccessEmails
-              bicAnalysis
-              investigatorEmail
-              investigatorName
-              isCmoRequest
-              labHeadEmail
-              labHeadName
-              libraryType
-              otherContactEmails
-              piEmail
-              projectManagerName
-              qcAccessEmails
-              smileRequestId
-            }
-            patientsHasSample {
-              smilePatientId
-              patientAliasesIsAlias {
-                namespace
-                value
-              }
-            }
-            cohortsHasCohortSample {
-              cohortId
-              hasCohortCompleteCohortCompletes {
-                date
-              }
-            }
-            hasTempoTempos {
-              smileTempoId
-              billed
-              billedBy
-              costCenter
-              custodianInformation
-              accessLevel
-              hasEventBamCompletes {
-                date
-                status
-              }
-              hasEventMafCompletes {
-                date
-                normalPrimaryId
-                status
-              }
-              hasEventQcCompletes {
-                date
-                reason
-                result
-                status
-              }
-            }
-          }`,
-        });
-
-        return samples;
+      async samples(
+        _source: undefined,
+        args: any,
+        { oncotreeCache }: ApolloServerContext
+      ) {
+        let customWhere = includeCancerTypeFieldsInSearch(
+          args.where,
+          oncotreeCache
+        );
+        return await querySamplesList(ogm, customWhere, args.options);
+      },
+      async samplesConnection(
+        _source: undefined,
+        args: any,
+        { oncotreeCache }: ApolloServerContext
+      ) {
+        let customWhere = includeCancerTypeFieldsInSearch(
+          args.where,
+          oncotreeCache
+        );
+        const samples = await querySamplesList(ogm, customWhere, args.options);
+        return {
+          totalCount: samples.length,
+        };
```

Comment:
Note that because we're calling `includeCancerTypeFieldsInSearch` 2x for each samples query, it's going to be 2x slower than before. We can discuss mitigating with the upcoming samples cache implementation.